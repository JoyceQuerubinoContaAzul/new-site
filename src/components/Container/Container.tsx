import { cn } from '@/utils/cn';
import type { ContainerProps } from './Container.types';

export const Container = (props: ContainerProps) => {
  const { children, className, ...rest } = props;

  return (
    <div className={cn('mx-auto max-w-[520px] px-5', className)} {...rest}>
      {children}
    </div>
  );
};
