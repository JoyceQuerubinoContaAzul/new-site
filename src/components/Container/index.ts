export { Container } from './Container';
export type * from './Container.types';
