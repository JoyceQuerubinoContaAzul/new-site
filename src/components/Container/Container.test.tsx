import { render } from '@/tests';
import { Container } from './Container';

type ContainerProps = React.ComponentProps<typeof Container>;

const setup = (props: Partial<ContainerProps> = {}) => {
  return render(<Container {...props} />);
};

describe('Container component', () => {
  it('should render the children', () => {
    const children = <div>Link</div>;

    const { getByText } = setup({ children });

    expect(getByText('Link')).toBeInTheDocument();
  });

  it('should render the expected structure', () => {
    const children = <div>Link</div>;

    const { container } = setup({ children });

    expect(container).toMatchSnapshot();
  });
});
