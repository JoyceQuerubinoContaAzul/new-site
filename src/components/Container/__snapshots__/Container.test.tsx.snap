// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Container component should render the expected structure 1`] = `
<div>
  <div
    class="mx-auto max-w-[520px] px-5"
  >
    <div>
      Link
    </div>
  </div>
</div>
`;
